src/frontend/js/components/OpenEdxFullNameForm/index.tsx

```diff
@@ -0,0 +1,165 @@
+import { Button, Input } from '@openfun/cunningham-react';
+import { FormattedMessage, defineMessages, useIntl } from 'react-intl';
+import { FormProvider, useForm } from 'react-hook-form';
+import * as Yup from 'yup';
+import { yupResolver } from '@hookform/resolvers/yup';
+import { useEffect, useState } from 'react';
+import { useSession } from 'contexts/SessionContext';
+import useOpenEdxProfile from 'hooks/useOpenEdxProfile';
+import Form, { getLocalizedCunninghamErrorProp } from 'components/Form';
+import { Spinner } from 'components/Spinner';
+import Banner, { BannerType } from 'components/Banner';
+import { UserHelper } from 'utils/UserHelper';
+
+const messages = defineMessages({
+  emailInputLabel: {
+    id: 'components.OpenEdxFullNameForm.emailInputLabel',
+    description: 'Label of "email" field of the openEdx full name form',
+    defaultMessage: 'Email',
+  },
+  fullNameInputLabel: {
+    id: 'components.OpenEdxFullNameForm.fullNameInputLabel',
+    description: 'Label of "fullName" field of the openEdx full name form',
+    defaultMessage: 'FullName',
+  },
+  fullNameInputDescription: {
+    id: 'components.OpenEdxFullNameForm.fullNameInputDescription',
+    description: 'Descripiton on the "fullName" field of the openEdx full name form.',
+    defaultMessage:
+      'Please check that your fullname is correct. It will be used on official document (e.g: certificate)',
+  },
+  submitButtonLabel: {
+    id: 'components.OpenEdxFullNameForm.submitButtonLabel',
+    description: 'Label of the submit button on openEdx full name form',
+    defaultMessage: 'Submit',
+  },
+  loadingLabel: {
+    id: 'components.OpenEdxFullNameForm.loadingLabel',
+    description: 'Label of the loading profile spinner on openEdx full name form',
+    defaultMessage: 'Loading...',
+  },
+  error: {
+    id: 'components.OpenEdxFullNameForm.error',
+    description: 'Message displayed i nopenEdx full name form when submit the request fail',
+    defaultMessage: 'There was an error while updating your full name. Please try again later.',
+  },
+});
+
+export interface OpenEdxFullNameFormValues {
+  name: string;
+}
+
+const validationSchema = Yup.object().shape({
+  name: Yup.string().required(),
+});
+
+const OpenEdxFullNameForm = () => {
+  const intl = useIntl();
+  const { user } = useSession();
+  const [isSuccess, setIsSuccess] = useState<boolean>(false);
+  const [isSubmitButtonDisabled, setIsSubmitButtonDisabled] = useState<boolean>(true);
+  const {
+    data: openEdxProfileData,
+    methods: { update },
+    states: { isFetched, isPending },
+    error,
+  } = useOpenEdxProfile({
+    username: user!.username,
+    onUpdateSuccess: () => {
+      setIsSuccess(true);
+    },
+  });
+
+  const defaultValues = {
+    name: (openEdxProfileData ? UserHelper.getName(openEdxProfileData) : '')?.trim(),
+  } as OpenEdxFullNameFormValues;
```

Comment:
Currently this value does not seems to be refreshed when user updates its fullname.